gym/f110_gym/envs/f110_env.py

```diff
@@ -229,10 +254,58 @@ def __del__(self):
         """
         pass
 
+    @classmethod
+    def default_config(cls) -> dict:
+        """
+        Default environment configuration.
+
+        Can be overloaded in environment implementations, or by calling configure().
+
+        Args:
+            None
+
+        Returns:
+            a configuration dict
+        """
+        return {
+            "seed": 12345,
+            "map_name": "vegas.yaml",
+            "map_ext": ".png",
+            "params": {
+                "mu": 1.0489,
+                "C_Sf": 4.718,
+                "C_Sr": 5.4562,
+                "lf": 0.15875,
+                "lr": 0.17145,
+                "h": 0.074,
+                "m": 3.74,
+                "I": 0.04712,
+                "s_min": -0.4189,
+                "s_max": 0.4189,
+                "sv_min": -3.2,
+                "sv_max": 3.2,
+                "v_switch": 7.319,
+                "a_max": 9.51,
+                "v_min": -5.0,
+                "v_max": 20.0,
+                "width": 0.31,
+                "length": 0.58,
+            },
+            "num_agents": 2,
+            "timestep": 0.01,
+            "ego_idx": 0,
+            "integrator": "rk4",
+            "control_input": "speed",
+        }
+
+    def configure(self, config: dict) -> None:
+        if config:
+            self.config.update(config)
```

Comment:
Replace `self.config.update(config)` with `pydantic.utils.deepupdate(self.config, config)`

Easier to change some parameters in nested dictionary without specifying all of them, such as in `config["params"]`.

Currently, the user has to pass all the simulation params. 
With `deepupdate` the user could give something like `config = {"params": {"mu": 2.0}}` and keep all the other parameters to the default value.

See example [https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth](https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth)

Reply:
Added deep_update from pydantic instead of adding a new dependency.